### ISSUE TITLE
loss exploding MPT

### ISSUE THREAD

**Issue by germanjke** [mosaicml/llm-foundry#232]
hi, i'm pretrain mpt 125m and after some tokens (arround 2200), loss starts exploding, do you have some issues on your pretraining?

right now i'm trying to use gradient clipping with biggest threshold, will see

**Comment by germanjke** [comment]
for example if set 5, it's going much better, but later (but later than 1.0 clip threshold) it's going to explode
maybe some more techniques?

**Comment by jfrankle** [comment]
Loss spikes can show up for a wide variety of reasons, and it's hard to say without looking closely at your hyperparameters and the specific examples in your dataset. It could be anything from hitting a bad batch in the dataset, the quality of the data overall, the learning rate, the amount of warmup, the weight decay, etc.

My initial recommendations would be to try lowering the learning rate, increasing the amount of warmup (since it's happening so early in training), and increasing the weight decay. Once you get into a stable regime of some kind, you can start gradually getting more aggressive with those hyperparameters. It's suspicious to me that it's happening so early in training...

We can't guarantee that training will be loss-spike-free with this codebase, so I'm going to close this issue for now. Please do let me know how these suggestions work out.
